storm/src/main/storm/mesos/MesosCommon.java

```diff
@@ -85,9 +92,29 @@ public static boolean startLogViewer(Map conf) {
   }
 
   public static int portFromTaskId(String taskId) {
-    int last = taskId.lastIndexOf("-");
-    String port = taskId.substring(last + 1);
-    return Integer.parseInt(port);
+    String[] parts = taskId.trim().split("-");
+    if (parts.length < 3) {
+      throw new IllegalArgumentException("TaskID " + taskId.trim() + " is invalid. " +
+          "Number of dash-delimited components (" + parts.length + ") is less than expected. " +
+          "Expected format is HOSTNAME-PORT-TIMESTAMP");
```

Comment:
There is no point in attempting to do that.  It's a brand new version of the framework, you *must* either launch a new framework or kill all existing executors.